lib/modules/manager/gomod/extract.spec.ts

```diff
@@ -19,8 +19,13 @@ describe('modules/manager/gomod/extract', () => {
       expect(res?.filter((e) => e.depType === 'replace')).toHaveLength(1);
     });
 
-    it('extracts constraints', () => {
+    it('extracts constraints and golang', () => {
       const res = extractPackageFile(gomod3);
+      const deps = res?.deps;
+      const filteredDeps = deps?.filter(
+        (dep) => dep.depType === 'golang' && dep.depName === 'go'
+      );
+      expect(filteredDeps).toHaveLength(1);
```

Comment:
You could use `toMatchObject` to streamline this 
```suggestion
      expect(res?.deps).toMatchObject([
        {
          depType: "golang",
          depName: "go",
        }
      ]);
```

Reply:
its a bit different than what you think, but i managed to find something in google, :D lets see

Reply:
Hmm thought it would allow partial arrays there.